composite/token-allocation/src/config.ts

```diff
@@ -21,24 +15,47 @@ enum DataProvider {
   Kaiko = 'kaiko',
 }
 
-const providers: Record<string, PriceAdapter> = {
-  [DataProvider.Amberdata]: amberdata,
-  [DataProvider.Cryptocompare]: cryptocompare,
-  [DataProvider.Coinpaprika]: coinpaprika,
-  [DataProvider.Nomics]: nomics,
-  [DataProvider.Coinmarketcap]: coinmarketcap,
-  [DataProvider.Coingecko]: coingecko,
-  [DataProvider.Coinapi]: coinapi,
-  [DataProvider.Kaiko]: kaiko,
+const providers: Record<string, DataProviderConfig> = {
+  [DataProvider.Amberdata]: {
+    batchingSupport: false,
+  },
+  [DataProvider.Cryptocompare]: {
+    batchingSupport: true,
+    batchEndpoint: 'multi',
+  },
+  [DataProvider.Coinpaprika]: {
+    batchingSupport: true,
+    batchEndpoint: 'multi',
+  },
+  [DataProvider.Nomics]: {
+    batchingSupport: true,
```

Comment:
Why did we focus on batching these requests? Is there a specific reason I am missing?

IMHO this introduces many new problems without really bringing any improvements. Batching is useful, but the way our system is currently designed is posing a problem.

Consider RL & warmed adapter, and an index token we need to price. If we don't batch, the adapter has a lot simpler state to work with. It tracks single data types, so when your index changes it can just start tracking the new data point, and with time close the subscription to the old one. All costs are 1, so by changing the index, we have a temporary dangling cost of M*1 (M tokens charged).

In the case of batching, we track the whole batch request. If your needs change by just 1 token, we need to start tracking the whole new batch and introduce a temporary dangling cost of N1 (size of the first batch).

So this is costing us a lot more in the end. Consider we timeout warmer subscriptions if we don't see a request for data in 2h. It can cost us up to (2/maxAge) * N to keep this dangling subscription.

We also don't reuse cached data between indexN1 & changed indexN2. We also don't reuse cached data between tokens listed in the index and feeds requesting the price of the token, standalone.

Not to mention code coupling and codebase problems this brings. What happens when Kaiko adds batching?

So in the end, batching will introduce a lot of inefficiency and cost, because of our current design, to save some free and fast HTTP round trips. I don't see it's worth it.

Reply:
Yep that makes sense, didn't consider warmer service here...

Reply:
I missed a case that I thought was nonexistent or very rare, but @boxhock just mentioned that API providers like CryptoCompare & CoinAPI count up to 100 in a batch as 1 RL token.

If that is the case then batching is valuable, but we still have the problems with it mentioned above that we need to figure out how to avoid.

Reply:
Well that's the reason behind all this batching. CMC also count them as 1.